### PR TITLE
#120: API-core-parallel_dispatch-parallel_for from md to rst

### PR DESCRIPTION
**Pull request by antoinemeyer5**
### THIS PR:
* API/core/parallel-dispatch/parallel_for transition from .md to .rst

before (.md) | after (.rst) | after v2 [**current code**] (.rst)
:----------------: | :-------------: | :-------------:
before_1 | after_1 | after2_1
before_2 | after_2 | after2_2
before_3 | after_3 | after2_3
before_4 | after_4 | after2_4

### CODE REVIEW

**Comment by antoinemeyer5** [comment]
Hello, I have a few questions and comments.

Questions : 
- isn't it weird to have "requirements:" in API/core/parallel-dispatch/parallel_for.md and to have "requirements" in API/core/initalize_finalize/finalize.rst? Shouldn't there be some consistency?

Overall I'm trying to make the documentation consistent: is this a good idea?

Remarks, for consistency I have:
- line 75 and 76: align Kokkos with main ( I did like here: https://kokkos.github.io/kokkos-core-wiki/API/core/initialize_finalize/finalize.html )
- line 3: I wrote `<Kokkos_Core.hpp>` to be consistent with here (link: https://kokkos.github.io/kokkos-core-wiki/API/core/initialize_finalize/finalize.html)

Tell me if the consistency is interesting or if I should stop and that it's a bad idea. I admit I took a little bit of liberty here so don't hesitate to tell me if it suits you. Best wishes.

**Comment by fnrizzi** [comment]
> * isn't it weird to have "requirements:" in API/core/parallel-dispatch/parallel_for.md and to have "requirements" in API/core/initalize_finalize/finalize.rst? Shouldn't there be some consistency?

I don't quite understand the question.... what is weird? the fact that both have *separate* requirements? 
If so, then no, it is not weird because finalize and parallel_for are totally independent functions. 


> * line 75 and 76: align Kokkos with main ( I did like here: https://kokkos.github.io/kokkos-core-wiki/API/core/initialize_finalize/finalize.html )

Don't worry about the alignment in this case, I would suggest just making sure the indentation is right. 
it is aligned or not does not matter 

> * line 3: I wrote `<Kokkos_Core.hpp>` to be consistent with here (link: https://kokkos.github.io/kokkos-core-wiki/API/core/initialize_finalize/finalize.html)

That is fine 

> Tell me if the consistency is interesting or if I should stop and that it's a bad idea. I admit I took a little bit of liberty here so don't hesitate to tell me if it suits you. Best wishes.

I think it is fine to keep an eye on that, but please priority converting the content

**Comment by ajpowelsnl** [comment]
@antoinemeyer5 -- I agree with @fnrizzi on your comments / questions, and prioritizing content over style at this time.

**Comment by antoinemeyer5** [comment]
> * isn't it weird to have "requirements:" in API/core/parallel-dispatch/parallel_for.md and to have "requirements" in API/core/initalize_finalize/finalize.rst? Shouldn't there be some consistency?

>> * I don't quite understand the question.... what is weird? the fact that they have *separate* requirements? 
If so, then no, it is not weird because finalize and parallel_for are different functions so they have different requirements.

It is strange to have one time the colon ':' and one time not the colon.

Okay I focus on the content more than on small details you are right @fnrizzi  and @ajpowelsnl : sorry to bother you with my silly questions. If everything is good for you and after checking, you can merge this branch.

**Comment by ajpowelsnl** [comment]
> > * isn't it weird to have "requirements:" in API/core/parallel-dispatch/parallel_for.md and to have "requirements" in API/core/initalize_finalize/finalize.rst? Shouldn't there be some consistency?
> 
> > > * I don't quite understand the question.... what is weird? the fact that they have _separate_ requirements?
> > >   If so, then no, it is not weird because finalize and parallel_for are different functions so they have different requirements.
> 
> It is strange to have one time the colon ':' and one time not the colon.
> 
> Okay I focus on the content more than on small details you are right @fnrizzi and @ajpowelsnl : sorry to bother you with my silly questions. If everything is good for you and after checking, you can merge this branch.

Morning @antoinemeyer5 -- Ne vous inquietez pas!  Not a silly question at all, just not the priority at the moment : )

**Comment by fnrizzi** [comment]
@ajpowelsnl you can remove the : at the end of requirements

File: docs/source/API/core/parallel-dispatch/parallel_for.rst [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    Kokkos::parallel_for(name, policy, functor);
+    Kokkos::parallel_for(policy, functor);
+
+.. _text: ../policies/ExecutionPolicyConcept.html
+
+.. |text| replace:: *ExecutionPolicy*
+
+Dispatches parallel work defined by ``functor`` according to the |text|_ ``policy``. The optional label ``name`` is
+used by profiling and debugging tools. This call may be asynchronous and return to the callee immediately. 
+
```
Comment (ajpowelsnl) [inline docs/source/API/core/parallel-dispatch/parallel_for.rst:22]: I know we're not dealing with content at this time, but the summary should also include `KOKKOS_LAMBDA`, in addition to the already mentioned `functor`.

File: docs/source/API/core/parallel-dispatch/parallel_for.rst [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+Parameters:
+~~~~~~~~~~~
+
+* ``name``: A user provided string which is used in profiling and debugging tools via the Kokkos Profiling Hooks. 
+* ExecPolicy: An *ExecutionPolicy* which defines iteration space and other execution properties. Valid policies are:
+
+  - ``IntegerType``: defines a 1D iteration range, starting from 0 and going to a count.
+  - `RangePolicy <../policies/RangePolicy.html>`_: defines a 1D iteration range.
+  - `MDRangePolicy <../policies/MDRangePolicy.html>`_: defines a multi-dimensional iteration space.
+  - `TeamPolicy <../policies/TeamPolicy.html>`_: defines a 1D iteration range, each of which is assigned to a thread team.
```
Comment (ajpowelsnl) [inline docs/source/API/core/parallel-dispatch/parallel_for.rst:48]: Future work (not required for merging now):  fix expression for `TeamPolicy`; should read something like, "...defines an execution policy for  1D iteration space starting at the begin iterate and going to the end iterate."

**Comment by antoinemeyer5** [comment]
> @antoinemeyer5 you can remove the : at the end of requirements also, can you please try to rewrite this part:
> 
> image
> 
> using Sphinx cpp syntax and render it so that we see what it looks like? You can ask @pierrepebay on how to do that since he is doing it for another page

Okay I'll talk to Pierre and I'll correct that. I have already made the change for the : at the ned of requirements. Thank you !

**Review by ajpowelsnl — APPROVED** [review]

**Comment by antoinemeyer5** [comment]
Result using ..cpp:function::

image

Visually it changes a lot.
It is impossible (or I haven't found it yet) to make the indentation clean.
@pierrepebay asked @fnrizzi for a little more explanation and he will tell me what is the best solution to get the best possible result.

**Review by fnrizzi — APPROVED** [review]

**Pull request merged**